### PR TITLE
Custom reporter GitHub integration and better formatting of special cases

### PR DESCRIPTION
**Pull request by alcuadrado**
This PR removes `@reporters/github` in favor of integrating directly ourselves from our reporter. 

While testing it I discovered two small formatting improvements that could be made, so I also included them here.

Note that there's a commit that includes an intentional test failure to test the integration. I'll remove it before merging.

### CODE REVIEW

File: v-next/hardhat-node-test-reporter/src/error-formatting.ts [hunk 1]
```diff
@@ -21,6 +23,14 @@ export function formatError(error: Error): string {
     );
   }
 
+  if (isTestFileExecutionFailureError(error)) {
```
Comment (schaable) [inline v-next/hardhat-node-test-reporter/src/error-formatting.ts:26]: 💯

File: v-next/hardhat-node-test-reporter/src/error-formatting.ts [hunk 2]
```diff
@@ -57,10 +68,9 @@ ${stack}`;
 
 // TODO: Do this in a more robust way and that works well with windows
```
Comment (schaable) [inline v-next/hardhat-node-test-reporter/src/error-formatting.ts:69]: is the TODO still needed?

File: v-next/hardhat-node-test-reporter/src/error-formatting.ts [hunk 3]
```diff
@@ -57,10 +68,9 @@ ${stack}`;
...
 function replaceFileUrlsWithRelativePaths(stack: string): string {
-  return stack.replaceAll(
-    "(" + pathToFileURL(process.cwd() + "/").toString(),
-    "(",
-  );
+  return stack
+    .replaceAll("(" + pathToFileURL(process.cwd() + path.sep).toString(), "(")
+    .replaceAll("(" + process.cwd() + path.sep, "(");
```
Comment (schaable) [inline v-next/hardhat-node-test-reporter/src/error-formatting.ts:73]: what's the difference with the previous replaceAll?

File: v-next/hardhat-node-test-reporter/src/github-actions.ts [hunk 4]
```diff
@@ -0,0 +1,63 @@
...
+  }
+
+  const error = event.details.error;
+
+  const location = getLocation(error);
+  if (location === undefined) {
+    return;
+  }
+
+  const { default: core } = await import("@actions/core");
```
Comment (schaable) [inline v-next/hardhat-node-test-reporter/src/github-actions.ts:25]: do we need to import packages dynamically in the reporter?

File: v-next/hardhat-node-test-reporter/src/error-formatting.ts (hunk 2 above)
Comment (alcuadrado) [reply v-next/hardhat-node-test-reporter/src/error-formatting.ts:69]: Yes, this is still a bit of a hack IMO. For example, see your later comment. Sometimes stack traces use file-urls, sometimes paths. Unclear when, and if we are properly handling everything. Eventually, we'll want to polish this, properly test it, etc.

File: v-next/hardhat-node-test-reporter/src/error-formatting.ts (hunk 3 above)
Comment (alcuadrado) [reply v-next/hardhat-node-test-reporter/src/error-formatting.ts:73]: See my previous comment

File: v-next/hardhat-node-test-reporter/src/github-actions.ts (hunk 4 above)
Comment (alcuadrado) [reply v-next/hardhat-node-test-reporter/src/github-actions.ts:25]: TBH it's unclear. Not very well documented by github if this package is safe outside of github actions. 

Also, locally it won't be used it, so it makes sense not to import it.

**Review by schaable — APPROVED** [review]

**Pull request merged**